src/qt/test/apptests.cpp

```diff
@@ -0,0 +1,113 @@
+#include "apptests.h"
+
+#include "chainparams.h"
+#include "init.h"
+#include "qt/bitcoin.h"
+#include "qt/bitcoingui.h"
+#include "qt/networkstyle.h"
+#include "qt/rpcconsole.h"
+#include "validation.h"
+
+#if defined(HAVE_CONFIG_H)
+#include "config/bitcoin-config.h"
+#endif
+#ifdef ENABLE_WALLET
+#include "wallet/db.h"
+#endif
+
+#include <QAction>
+#include <QEventLoop>
+#include <QLineEdit>
+#include <QScopedPointer>
+#include <QTest>
+#include <QTextEdit>
+#include <QtGlobal>
+#if QT_VERSION >= 0x050000
+#include <QtTest/QtTestWidgets>
+#endif
+#include <QtTest/QtTestGui>
+#include <new>
+#include <string>
+#include <univalue.h>
+
+namespace {
+//! Call getblockchaininfo RPC and check first field of JSON output.
+void TestRpcCommand(RPCConsole* console)
+{
+    QEventLoop loop;
+    QTextEdit* messagesWidget = console->findChild<QTextEdit*>("messagesWidget");
+    QObject::connect(messagesWidget, SIGNAL(textChanged()), &loop, SLOT(quit()));
+    QLineEdit* lineEdit = console->findChild<QLineEdit*>("lineEdit");
+    QTest::keyClicks(lineEdit, "getblockchaininfo");
+    QTest::keyClick(lineEdit, Qt::Key_Return);
+    loop.exec();
+    QString output = messagesWidget->toPlainText();
+    UniValue value;
+    value.read(output.right(output.size() - output.lastIndexOf(QChar::ObjectReplacementCharacter) - 1).toStdString());
+    QCOMPARE(value["chain"].get_str(), std::string("regtest"));
+}
+} // namespace
+
+//! Entry point for BitcoinApplication tests.
+void AppTests::appTests()
+{
+#ifdef Q_OS_MAC
+    if (QApplication::platformName() == "minimal") {
+        // Disable for mac on "minimal" platform to avoid crashes inside the Qt
+        // framework when it tries to look up unimplmented cocoa functions, and
+        // fails to handle returned nulls
+        // (https://bugreports.qt.io/browse/QTBUG-49686).
+        QWARN("Skipping AppTests on mac build with 'minimal' platform set due to Qt bugs. To run AppTests, invoke "
+              "with 'test_bitcoin-qt -platform cocoa' on mac, or else use a linux or windows build.");
+        return;
+    }
+#endif
+
+    m_app.parameterSetup();
+    m_app.createOptionsModel(true /* reset settings */);
+    QScopedPointer<const NetworkStyle> style(
+        NetworkStyle::instantiate(QString::fromStdString(Params().NetworkIDString())));
+    m_app.createWindow(style.data());
+    BitcoinCore::baseInitialize();
+    connect(&m_app, SIGNAL(windowShown(BitcoinGUI*)), this, SLOT(guiTests(BitcoinGUI*)));
+    expectCallback("guiTests");
+    m_app.requestInitialize();
+    m_app.exec();
+
+    // Reset some global state to avoid interfering with later tests.
+    ResetShutdownRequested();
+#ifdef ENABLE_WALLET
+    ::bitdb.Close();
```

Comment:
Should `bitdb` be called directly? See https://github.com/bitcoin/bitcoin/pull/9951/commits/be9e1a968debbb7ede8ed50e9288a62ff15d1e1e

Reply:
> Should bitdb be used directly? See be9e1a9

Probably these calls should just be dropped. I had to make similar changes to wallettests.cpp 
in #11687 (which was merged since the PR was opened): https://github.com/bitcoin/bitcoin/pull/11687/commits/d8a99f65e53019becdd8d2631396012bafb29741#diff-6fcb1829ce66ad65046d6896dbcc910b

Reply:
If I remove those lines `make` is happy (but tests still fail, see below).